graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/admin/DeploySchemaFetcherBase.java

```diff
@@ -77,40 +78,42 @@
     MigrationStrategy migrationStrategy = environment.getArgument("migrationStrategy");
     boolean dryRun = environment.getArgument("dryRun");
 
-    ImmutableMap.Builder<String, Object> response = ImmutableMap.builder();
-    ProcessedSchema processedSchema =
-        new SchemaProcessor(authenticationService, authorizationService, dataStoreFactory)
-            .process(input, keyspace);
-    response.put(
-        "messages",
-        processedSchema.getMessages().stream()
-            .map(this::formatMessage)
-            .collect(Collectors.toList()));
+    return schemaSourceDao.tryDeployNewSchema(
+        namespace,
+        () -> {
+          ImmutableMap.Builder<String, Object> response = ImmutableMap.builder();
+          ProcessedSchema processedSchema =
+              new SchemaProcessor(authenticationService, authorizationService, dataStoreFactory)
+                  .process(input, keyspace);
+          response.put(
+              "messages",
+              processedSchema.getMessages().stream()
+                  .map(this::formatMessage)
+                  .collect(Collectors.toList()));
```

Comment:
Per my other comment int the DAO, I think we should handle errors here instead of passing a callable. Another reason is that if `dryRun` is set we don't want to write anything to the database:

```java
    List<MigrationQuery> queries;
    SchemaSourceDao schemaSourceDao = new SchemaSourceDao(dataStore);
    if (!dryRun) {
      schemaSourceDao.startDeployment(); // sets the flag with LWT
    }
    try {
      ProcessedSchema processedSchema = ...
      response.put( "messages", ...
      queries = ...
      response.put("cqlChanges", ...
    } catch (Exception e) {
      if (!dryRun) {
        schemaSourceDao.abortDeployment(); // unsets the flag (no need for LWT)
      }
      throw e;
    }
    if (!dryRun) {
      for (MigrationQuery query : queries) { ... }
      // inserts the new version **and** unsets the flag (no need for LWT)
      SchemaSource newSource = schemaSourceDao.insert(namespace, expectedVersion, input);
      response.put("version", newSource.getVersion());
    }
```

Note that I'm also suggesting that `insert` should unset `deployment_in_progress`. It can all be done in one query now, we don't need an LWT or a CQL batch anymore:
```
INSERT INTO graphql_schema(key, version, latest_version, contents, deployment_in_progress)
VALUES ('key', <new_version>, <new_version>, 'the contents...', false)
```

Reply:
Thank you for this comment, very good ideas!.
I've addressed the feedback. Two additional things that I've added:
1. We need to check if `schemaSourceDao.startDeployment()` was successful, and throw if not:
```
if (!schemaSourceDao.startDeployment(namespace, expectedVersion)) {
        throw new IllegalStateException(
            String.format(
                "It looks like someone else is deploying a new schema for namespace: %s and version: %s. "
                    + "Please retry later.",
                namespace, expectedVersion));
      }
``` 
otherwise, the deployment will progress regardless of the fact if there was `deployment_in_progress` or not
2. We don't need to pass the lastest_schema_version to insert anymore:
```
 SchemaSource newSource = schemaSourceDao.insert(namespace, input);
```
because it inserts a new version for both `version` and `latest_version` columns. 
Do you agree with those changes?

Reply:
Yes 👍